src/pages/program/[slug]/mentors/index.tsx

```diff
@@ -24,20 +27,53 @@ const ViewMentorsPage: Page = () => {
       profileType: ProfileType.Mentor,
     },
   });
+  const { data: profileTagsData } = useGetProfileTagsByProgramQuery({
+    variables: { programId: currentProgram?.programId! },
+  });
   let unfiltered = data?.getProfiles || [];
+  const [mentors, setMentors] = useState(unfiltered);
+
+  useEffect(() => {
+    if (unfiltered) {
+      setMentors(unfiltered);
+    }
+  }, [unfiltered]);
 
   const fuse = new Fuse(unfiltered, {
     keys: ["user.firstName", "user.lastName", "profileJson"],
   });
-  const mentors = searchText
-    ? fuse.search(searchText).map((x) => x.item)
-    : unfiltered;
+
+  const filterMentors = (tags: string[]) => {
```

Comment:
This function doesn't need to set state again--we can simply use `allMentors`, `Fuse`, and `profileTagIds` to calculate the filtered mentors

Reply:
I'm a little confused on what this means- do you mean to just return the filtered mentors instead of setting state?